great_expectations/experimental/datasources/interfaces.py

```diff
@@ -396,6 +398,7 @@ def add_asset(self, asset: DataAssetType) -> DataAssetType:
         # The setter for datasource is non-functional, so we access _datasource directly.
         # See the comment in DataAsset for more information.
         asset._datasource = self
+        asset.test_connection()
```

Comment:
Happy to discuss adding a `test_connection` arg to `add_asset` after consulting with product.